torch/csrc/jit/runtime/autodiff.cpp

```diff
@@ -356,6 +356,28 @@ static Value* createAutogradAdd(Value* a, Value* b) {
   return graph->insertNode(graph->create(prim::AutogradAdd, {a, b}))->output();
 }
 
+namespace {
+bool outputRequiresGrad(Value* output) {
+  if (output->type()->castRaw<TensorType>() == nullptr) {
+    return output->requires_grad();
+  }
+  c10::optional<bool> requiresGrad =
+      output->type()->expectRef<TensorType>().requiresGrad();
+  if (requiresGrad.has_value()) {
+    return *requiresGrad;
+  }
+
+  Node* n = output->node();
+  if (n->kind() != prim::profile) {
+    return true;
+  }
+  if (!n->hasAttribute(attr::profiled_type)) {
```

Comment:
We profile on use not on def right, so I guess the output is always `prim::profile`

Reply:
Could you try something with multiple uses that can't be merged in what group in a test ?

Reply:
Added fixes in https://github.com/pytorch/pytorch/pull/78875